Homework/2/Program.cs

```diff
@@ -0,0 +1,470 @@
+
+
+        // Returns true if username is found
+        static bool CheckIfUserExists(string username) {
+            for (int i = 0; i < userArray.Length; i++)
+            {
+                if (userArray[i][0].ToLower() == username.ToLower())
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        // Look through user array for a username and return string array of username and password
+        static string[] ReturnUserByUsername(string username) {
+            string[] array = new string[] {"",""};
+            for (int i = 0; i < userArray.Length; i++)
+            {
+                if (userArray[i][0].ToLower() == username.ToLower())
+                {
+                    array = userArray[i];
+                }
+            }
+            return array;
+        }
+
+        //Adding string[] - [username,password] - user details to userArray at selected index.
+        static void AddUserToArrayAtIndex(string[] userDetails, int index) {
+            string[][] newArray = new string[userArray.Length + 1][];
+
+            for (int i = 0; i < newArray.Length; i++)
+            {
+                if (index == 0 && i == 0 )
+                {
+                    newArray[i] = userDetails;
+                }else if(i < index)
+                {
+                    newArray[i] = userArray[i];
+                }else if (i == index)
+                {
+                    newArray[i] = userDetails;
+                }else if (i > index)
+                {
+                    newArray[i] = userArray[i - 1];
+                }
+            }
+
+            userArray = newArray;
+        }
+
+        //Function to list users;
+        static void ListUsers() {
+            Console.Clear();
+            Console.WriteLine("Users:\n");
+
+            for (int i = 0; i < userArray.Length; i++)
+            {
+                Console.WriteLine(userArray[i][0]);
+            }
+            PromptInput("\nPress any key to continue.");
+            MenuSelection();
+        }
+
+        //Function to add userDetails to end or start of userArray. Taking bool variables to determine whether add to start or end or select index.
+        static void PromptAddUserToArray(bool addToEnd = true,bool addToStart = false) {
+            Console.Clear();
+            Console.WriteLine("Adding user:\n");
+
+            string username = PromptInput("Enter username:");
+            string password = PromptInput("Enter password:");
+
+
+            int index = GetIndex(addToEnd, addToStart);
+
+            if (!CheckIfUserExists(username) && !(password == "") && index != -1)
+            {
+                string[] userDetails = new string[2] { username, password };
+                AddUserToArrayAtIndex(userDetails, index);
+                WriteUsersToFile();
+                ListUsers();
+            }
+            else {
+                if (RetryConfirmation("Bad user details."))
```

Comment:
Well, it might be a bad position and good user details.  Also, what does "bad user details" mean?

Reply:
It's just some default error message, I didn't want to deal with all false cases.